dspace-oai/src/main/java/org/dspace/event/OAIIndexEventConsumer.java

```diff
@@ -0,0 +1,169 @@
+ */
+public class OAIIndexEventConsumer implements Consumer {
+    /**
+     * log4j logger
+     */
+    private static Logger log = Logger.getLogger(OAIIndexEventConsumer.class);
+
+    ItemService itemService = ContentServiceFactory.getInstance().getItemService();
+
+    // collect Items, Collections, Communities that need indexing
+    private Set<Item> itemsToUpdate = null;
+
+    @Override
+    public void initialize() throws Exception {
+        // No-op
+    }
+
+    /**
+     * Consume a content event -- just build the sets of objects to add (new) to
+     * the index, update, and delete.
+     *
+     * @param ctx   DSpace context
+     * @param event Content event
+     */
+    public void consume(Context ctx, Event event) throws Exception {
+
+        if (Objects.isNull(itemsToUpdate)) {
+            itemsToUpdate = new HashSet<Item>();
+        }
+
+        int st = event.getSubjectType();
+        if (!(st == Constants.ITEM || st == Constants.BUNDLE
+                || st == Constants.COLLECTION || st == Constants.COMMUNITY || st == Constants.BITSTREAM)) {
+            log
+                    .warn("IndexConsumer should not have been given this kind of Subject in an event, skipping: "
+                            + event.toString());
+            return;
+        }
+
+        DSpaceObject subject = event.getSubject(ctx);
+        DSpaceObject object = event.getObject(ctx);
+
+        int et = event.getEventType();
+
+        if (Objects.nonNull(object) && event.getObjectType() == Constants.ITEM) {
+            //just update the object
+            itemsToUpdate.add((Item)object);
+            return;
+        }
+
+        if (Objects.isNull(subject)) {
+            return;
+        }
+
+        if (event.getSubjectType() == Constants.COLLECTION || event.getSubjectType() == Constants.COMMUNITY) {
+            if (et == Event.MODIFY || et == Event.MODIFY_METADATA || et == Event.REMOVE || et == Event.DELETE) {
+                //must update all the items
+                if (subject.getType() == Constants.COMMUNITY) {
+                    for (Collection col : ((Community)subject).getCollections()) {
+                        addAll(ctx, col);
+                    }
+                } else {
+                    addAll(ctx, (Collection)subject);
+                }
+            }
+        } else if (event.getSubjectType() == Constants.BITSTREAM || event.getSubjectType() == Constants.BUNDLE) {
+            //must update owning items regardless the event
+            if (subject.getType() == Constants.BITSTREAM) {
+                for (Bundle bun : ((Bitstream)subject).getBundles()) {
+                    itemsToUpdate.addAll(bun.getItems());
+                }
+            } else {
+                itemsToUpdate.addAll(((Bundle)subject).getItems());
+            }
+        } else if (event.getSubjectType() == Constants.ITEM) {
+            //any event reindex this item
```

Comment:
Your comments are not consistent. In some places, you add a space at the beginning of the comment, and in others, you don't. Somewhere you start with the uppersase somewhere not.
I suggest to start with uppercase and add a space before the message.